assets/js/util.js

```diff
@@ -242,8 +243,26 @@ function synchronizeTextareaHeights() {
     $('#translatedText').css('height', originalTextScrollHeight + 'px');
 }
 
-/*:: export {synchronizeTextareaHeights, modeEnabled, ajaxSend, ajaxComplete, filterLangList, onlyUnique,
-    SPACE_KEY_CODE, ENTER_KEY_CODE, HTTP_OK_CODE, HTTP_BAD_REQUEST_CODE, XHR_LOADING, XHR_DONE} */
+function apyCall(requestObject, endpoint) {
+    window[requestObject.requestName] = $.ajax({
```

Comment:
Ohh! Um, something like the following:

```
function callApy(options, endpoint) {
    var targetObj = Object.assign({
        url: config.APY_URL + endpoint,
        beforeSend: ajaxSend,
        contentType: 'application/x-www-form-urlencoded; charset=UTF-8',
        dataType: 'json'
    },
    {
        data: options.request,
        type: (options.request.q.length > THRESHOLD_REQUEST_LENGTH) ? 'POST' : 'GET',
        complete: function () {
            ajaxComplete();
            window[options.requestName] = undefined;
        },
        success: options.success,
        error: options.error
    });

    window[options.requestName] = $.ajax(targetObj);
}
```

Here the parameters that are common across all the callees go into `target` by default. The remaining params go as a part of the `options` values in the `source`.